projects/ion/src/lib/step/step.component.ts

```diff
@@ -0,0 +1,139 @@
+} from '@angular/core';
+
+export interface StepType {
+  label: string;
+  description?: string;
+  index?: number;
+  status?: 'default' | 'selected' | 'checked' | 'error';
+}
+
+export type StepTheme = 'default' | 'pastel-tones' | 'dark-tones';
+
+export type CrossType = 'initial' | 'final';
+
+@Component({
+  selector: 'ion-step',
+  templateUrl: './step.component.html',
+  styleUrls: ['./step.component.scss'],
+})
+export class StepComponent implements OnInit, OnChanges {
+  @Input() current = 1;
+  @Input() disabled = false;
+  @Input() theme: StepTheme = 'default';
+  @Input() steps: StepType[];
+  @Output() indexChange = new EventEmitter<number>();
+
+  public indexShowed = 1;
+
+  public startSteps(): void {
+    let index = 1;
+    const firstStep = this.steps[0];
+
+    firstStep.status = firstStep.status ? firstStep.status : 'selected';
+
+    this.steps.map((step, indexArray) => {
+      const previusStep = this.steps[indexArray - 1];
+      const actualStepStatus = step.status;
+      if (
+        actualStepStatus !== 'checked' &&
+        previusStep &&
+        previusStep.status === 'checked'
+      ) {
+        step.status = actualStepStatus ? actualStepStatus : 'selected';
+      } else {
+        step.status = actualStepStatus ? actualStepStatus : 'default';
+      }
+      step.index = index;
+      ++index;
+    });
+  }
+
+  public crossVisibility(crossType: CrossType, step: StepType): boolean {
+    const condition =
+      crossType === 'initial'
+        ? step.index === 1
+        : step.index === this.steps.length;
+
+    if (condition) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  public crossClassGenerator(crossType: CrossType, index: number): string {
+    const condition =
+      crossType === 'initial'
+        ? this.steps[index - 1] && this.steps[index - 1].status === 'checked'
+        : this.steps[index + 1] && this.steps[index + 1].status === 'checked';
+
+    if (this.steps[index].status === 'checked' || condition) {
+      return 'cross bolded';
+    } else {
+      return 'cross';
+    }
+  }
+
+  public disabledConcactClass(prefix: string): string {
+    return this.disabled ? prefix + ' disabled' : prefix;
+  }
+
+  public labelClassGenerator(step: StepType): string {
+    if (step.status === 'selected' || step.status === 'error') {
+      return this.disabledConcactClass('label selected');
+    } else {
+      return this.disabledConcactClass('label');
+    }
+  }
+
+  public descriptionClassGenerator(): string {
+    return this.disabledConcactClass('description');
+  }
+
+  public reorganizeSteps(): void {
+    this.steps.map((step, indexArray) => {
+      const indexComparativo = indexArray + 1;
+      if (indexComparativo < this.indexShowed) {
+        step.status = 'checked';
+      } else if (indexComparativo === this.indexShowed) {
+        step.status = 'selected';
+      } else if (indexComparativo > this.indexShowed) {
+        step.status = 'default';
```

Comment:
```suggestion
        step.status = ENUM.CHECKED;
      } else if (indexComparativo === this.indexShowed) {
        step.status = 'selected';
      } else if (indexComparativo > this.indexShowed) {
        step.status = 'default';
```

Use Enum to compare this strings